src/main/java/hudson/plugins/plot/CSVSeries.java

```diff
@@ -109,36 +109,55 @@ public boolean getDisplayTableFlag() {
     @Override
     public List<PlotPoint> loadSeries(FilePath workspaceRootDir,
                                       int buildNumber, PrintStream logger) {
+        List<PlotPoint> ret = null;
+
+        FilePath[] seriesFiles;
+        try {
+            seriesFiles = workspaceRootDir.list(getFile());
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Exception trying to retrieve series files", e);
+            return null;
+        }
+
+        if (ArrayUtils.isEmpty(seriesFiles)) {
+            LOGGER.info("No plot data file found: " + workspaceRootDir.getName()
+                    + " " + getFile());
+            return null;
+        }
+
+        for (int index = 0; index < seriesFiles.length; index++) {
```

Comment:
how about `foreach` syntax instead?
```suggestion
        for (FilePath seriesFile : seriesFiles) {
            List<PlotPoint> seriesList = loadSeriesFile(seriesFile, buildNumber, logger);
```